src/Compiler/SyntaxTree/SyntaxTree.fsi

```diff
@@ -1046,12 +1046,36 @@ type SynSimplePats =
 
     member Range: range
 
+/// Represents a single named argument pattern a pair of the form `name = pattern`.
+[<NoEquality; NoComparison>]
+type NamePatPairField =
+    | NamePatPairField of
+        /// The identifier of the named field/parameter.
+        fieldName: SynLongIdent *
+        /// The range of the equals sign in `name = pattern`, if present.
+        equalsRange: range option *
+        /// The overall range of this name–pattern pair.
+        range: range option *
```

Comment:
Could you please write invariants about ranges in the comment?
It would help me (and perharps others) to better reason about the data.

is beginning of fieldName.Range the same as beginning of range?
When is `range` missing?

Is the end of pat.Range the same as the end of range?

Reply:
Great, thanks